File: Test/Test-Fibonacci/java/Fibonacci.java

public class Fibonacci {
	public static void main(String[] args) {
		long ns = System.nanoTime();
		new Fibonacci().start();
		long time = (System.nanoTime() - ns)/1000000;
		System.out.println(time + "ms");
	}
	
	public void start() {
		fibonacci(40);
	}
	
	public long fibonacci(int n) {
		if (n == 0)
			return 0;
		else if (n == 1)
			return 1;
		else
			return fibonacci(n-1) + fibonacci(n-2);
	}
}

File: Test/Test-Fibonacci/cs-mono/TestFibonacci/TestFibonacci/Fibonacci.cs
using System;
using System.Diagnostics;

namespace TestFibonacci
{
	public class Fibonacci
	{
		static void Main() {
			//Console.WriteLine(GC.GetTotalMemory (true));
			Console.WriteLine(System.Diagnostics.Process.GetCurrentProcess ().WorkingSet64);
			new Fibonacci ().start();
			Console.WriteLine(System.Diagnostics.Process.GetCurrentProcess ().WorkingSet64);
			Console.WriteLine(GC.GetTotalMemory (true));
		}

		public void start() {
			fibonacci (40);
		}

		public long fibonacci(int n) {
			if (n == 0)
				return 0;
			else if (n == 1)
				return 1;
			else
				return fibonacci (n - 1) + fibonacci (n - 2);
		}
	}
}

